### PR TITLE
build: docker-compose for dev

### PR DESCRIPTION
**Pull request by alexgarel**
Dockerfile and docker-compose to have a prod and dev setup for frontend

fixes https://github.com/openfoodfacts/taxonomy-editor/issues/56

### CODE REVIEW

**Comment by alexgarel** [comment]
@aadarsh-ram seems close but when I try to go to the add at http://ui.taxonomy.localhost:8091/ the page is blank !

**Comment by alexgarel** [comment]
@aadarsh-ram should work now :-)

**Review by teolemon — APPROVED** [review]
@aadarsh-ram feel free to review as well

**Review by aadarsh-ram — COMMENTED** [review]
@alexgarel I made some simple text changes.

File: doc/introduction/setup-dev.md [hunk 1]
```diff
@@ -0,0 +1,28 @@
+# Setup a dev environment
+
+## Using docker
+
+The docker-compose is already ok for dev.
+
+So using:
+
+```bash
+docker-compose build
```
Comment (aadarsh-ram) [inline doc/introduction/setup-dev.md:10]:
```suggestion
DOCKER_BUILDKIT=1 docker-compose build
```

Comment (aadarsh-ram) [inline doc/introduction/setup-dev.md:10]:
```suggestion
DOCKER_BUILDKIT=1 docker-compose build
```

**Review by aadarsh-ram — CHANGES_REQUESTED** [review]
@alexgarel I tested this setup and I got this error:
```
Step 7/15 : RUN --mount=type=cache,id=taxonomy-editor-npm-cache,target=/home/node/.npm,mode=0777   chown -R node:node /home/node/.npm
the --mount option requires BuildKit. Refer to https://docs.docker.com/go/buildkit/ to learn how to build images with BuildKit enabled
ERROR: Service 'taxonomy_node' failed to build : Build failed
```
After adding `DOCKER_BUILDKIT=1` before the build command, I'm able to view the website. The build command takes a long time and a comment about it in `setup-dev.md` will be appreciated. If there's a way to reduce the build time, please do look into it.

File: doc/introduction/setup-dev.md (hunk 1 above)
Comment (alexgarel) [reply doc/introduction/setup-dev.md:10]: hum if we need DOCKER_BUILDKIT, I will add it to the .env

File: doc/introduction/setup-dev.md [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+docker-compose up
+```
+
+You should have a running environment.
+
+The react application is exposed at: https://ui.taxonomy.localhost:8091
+The api is exposed at: https://api.taxonomy.localhost:8091
```
[the remark this answers is not in the document]
Comment (alexgarel) [reply doc/introduction/setup-dev.md:17]: you're right :-)

File: doc/introduction/setup-dev.md (hunk 1 above)
Comment (alexgarel) [reply doc/introduction/setup-dev.md:10]: oh no, we cannot, ok let's document then.

**Comment by alexgarel** [comment]
@aadarsh-ram : I modified the documentation, is that ok for you.

The build is only necessary the first time, or on changes that affects the setup (new packages, etc.)

**Review by aadarsh-ram — APPROVED** [review]
LGTM

**Pull request merged**